CRM/Contribute/BAO/ContributionPage.php

```diff
@@ -26,17 +26,12 @@
 class CRM_Contribute_BAO_ContributionPage extends CRM_Contribute_DAO_ContributionPage implements HookInterface {
 
   /**
-   * Creates a contribution page.
-   *
+   * @deprecated
    * @param array $params
-   *
    * @return CRM_Contribute_DAO_ContributionPage
    */
   public static function create($params) {
-    // @todo  - this implode is probably handled by writeRecord - test & remove.
-    if (isset($params['payment_processor']) && is_array($params['payment_processor'])) {
-      $params['payment_processor'] = implode(CRM_Core_DAO::VALUE_SEPARATOR, $params['payment_processor']);
```

Comment:
Indeed it is handled by writeRecord. And there is a test:
https://github.com/civicrm/civicrm-core/blob/4772bbd375c116226cd23a38a423fdf8c576fc71/tests/phpunit/api/v3/ContributionPageTest.php#L224-L226